tests/utils/select-block-by-name.js

```diff
@@ -0,0 +1,11 @@
+/**
+ * External dependencies
+ */
+import { getAllBlocks, selectBlockByClientId } from '@wordpress/e2e-test-utils';
+
+export const selectBlockByName = async ( name ) => {
+	await selectBlockByClientId(
+		( await getAllBlocks() ).find( ( block ) => block.name === name )
+			.clientId
+	);
+};
```

Comment:
This could would only work for top-level blocks, not innerBlocks ones since those are children of `getAllBlocks`.
This leet code should work:

```js
export const selectBlockByName = async ( blockName ) => {
	const blocksInEditor = await getAllBlocks();
	const flatBlockArray = ( blocks ) =>
		blocks
			.map( ( block ) => [
				{ [ block.name ]: block.clientId },
				block.innerBlocks ? flatBlockArray( block.innerBlocks ) : [],
			] )
			.flat( Infinity );
	const blocksObject = Object.fromEntries(
		flatBlockArray( blocksInEditor )
			.map( ( block ) => Object.entries( block ) )
			.flat()
	);
	const clientId = blocksObject[ blockName ];
	return selectBlockByClientId( clientId );
};
```